helm/orca/config/alerts-mapping.yaml

```diff
@@ -185,6 +185,416 @@ prometheus:
         kind: node
         properties:
           name: instance
+    - name: KubeCronJobRunning
+      source_mapping:
+        origin: kubernetes
+        kind: cronjob
+        properties:
+          name: cronjob
+          namespace: namespace
+    - name: KubeJobCompletion
+      source_mapping:
+        origin: kubernetes
+        kind: job
+        properties:
+          name: job_name
+          namespace: namespace
+    - name: KubeJobFailed
+      source_mapping:
+        origin: kubernetes
+        kind: job
+        properties:
+          name: job_name
+          namespace: namespace
+    - name: KubeAPIDown
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: KubeSchedulerDown
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: KubeControllerManagerDown
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: KubeCPUOvercommit
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: KubeMemOvercommit
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: KubeMemoryOvercommit
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: KubeCPUQuotaOvercommit
+      source_mapping:
+        origin: kubernetes
+        kind: namespace
+        properties:
+          name: namespace
+    - name: KubeMemoryQuotaOvercommit
+      source_mapping:
+        origin: kubernetes
+        kind: namespace
+        properties:
+          name: namespace
+    - name: KubeQuotaExceeded
+      source_mapping:
+        origin: kubernetes
+        kind: namespace
+        properties:
+          name: namespace
+    - name: KubeVersionMismatch
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: KubeStateMetricsDown
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: AlertmanagerDown
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: NodeExporterDown
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: PrometheusDown
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
+    - name: PrometheusOperatorDown
+      source_mapping:
+        origin: kubernetes
+        kind: cluster
+        properties: {}
```

Comment:
I'm a little bit hesitant about mapping these alerts to the cluster kind. They are not related to Kubernetes cluster, but rather Prometheus jobs/services, and as such may disturb the analytical field of view.

Alerts such as `PrometheusOperatorDown` or `NodeExporterDown` indicate that Prometheus instance was unable to scrape job endpoints (unable to fetch metrics). As such, similar alerts should be mapped to e.g. Prometheus service object that prevents metric scrape. Alternatively, we could map to Prometheus instances (pods) which reported job issues.

In the future we should think of a different mapping solution for this category of alerts. If you see other options that we could implement in this change, please comment. Otherwise, let's keep the mapping as it is and add a separate issue for mapping improvements.

Reply:
Unfortunately, these are the only suggested changes that I couldn't make.
Alerts with "* Down" use absent() function, which doesn't return any labels ("returns an empty vector if the vector passed to it has any elements and a 1-element vector with the value 1 if the vector passed to it has no elements"), so I'll create a separate issue for mapping improvements.